File: 10_Days_of_Statistics/Pearson_coefficient.py
import math

def mu(a, n):
    sum = 0
    for i in range(n):
        sum += a[i]
    return sum/n

def std(a, n, mu):
    sum = 0
    for i in range(n):
        sum += (a[i] - mu)**2
    return math.sqrt(sum/n)
        
def corr(n, x, y):
    sum = 0
    mu_x = mu(x, n)
    mu_y = mu(y, n)
    for i in range(n):
        sum += (x[i] - mu_x)*(y[i] - mu_y)
    return sum/(n*(std(x, n, mu_x)*std(y, n, mu_y)))



n = int(input())
x = list(map(float, input().strip().split(" ")))
y = list(map(float, input().strip().split(" ")))

print(round(corr(n, x, y), 3))

File: 10_Days_of_Statistics/Pearson_coefficient.cpp
#include <iostream>
#include <math.h>
#include <iomanip>
#include <vector>


using namespace std;

float mu(const vector<float> &a, int n){
    float sum = 0;
    for(int i = 0; i<n; i++){
        sum += a[i];
    }
    return sum/n;
}

float std_dev(const vector<float> &a, int n, float mu){
    float sum1 = 0;
    for(int i=0; i<n; i++){
        sum1 += pow((a[i] - mu), 2);
    }
    return pow((sum1/n), 0.5);
}

float corr(int n, const vector<float> &x, const vector<float> &y){
    float mu_x = mu(x, n);
    float mu_y = mu(y, n);
    float sum = 0;
    for(int i = 0; i<n; i++){
        sum += (x[i] - mu_x)*(y[i] - mu_y);
    }
    return sum/(n*std_dev(x, n, mu_x)*std_dev(y, n, mu_y));
}

int main() {
    int n, i;
    cin >> n;
    vector<float> x(n);
    for(i = 0; i<n; i++){
        cin >> x[i];
    }
    vector<float> y(n);
    for(i = 0; i<n; i++){
        cin >> y[i];
    }

    cout << std::fixed << std::setprecision(3) << corr(n, x, y) << endl;

   return 0;
}
